src/content/en/2021/caching.md

```diff
@@ -17,4 +20,470 @@ featured_stat_3: TODO
+
+This higher adoption of Service Workers from the top 1k pages, could be related to the world-wide trend towards remote-first and by association, mobile-friendly. As our reliance on working and living in one place throughout the entire year shifts, we need our devices to work even harder and smarter to keep up with us. Service Workers are a tool that can improve performance when the user is dealing with unreliable networks or LowEnd devices.
+
+{{ figure_markup(
+  image="sw-adoption.png",
+  caption="Service Worker adoption.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=688062455&format=interactive",
+  sheets_gid="802609299",
+  sql_file="service_worker_rank.sql"
+) }}
+
+The primary way to cache resources within a Service Worker is by using the *CacheStorage API*. This allows a developer to create a custom cache strategy for any requests passing through the worker; some well-known ones are *stale-while-revalidate*, *Cache Falling Back to Network*, *Network Falling Back to Cache*, and *Cache Only*. In recent years it has become even easier to adopt those strategies thanks to the increased popularity of [Workbox](https://developers.google.com/web/tools/workbox/modules/workbox-strategies), which helps you decide what cache you want to plug and play.
+
+## Headers adoption
+
+With both a CDN and the Browser, HTTP Headers are the primary toolkit a developer must master to properly cache resources. Headers are simply instructions read during the HTTP Request or Response, and some of them help control the cache strategy used.
+
+The caching-related headers, or the absence of them, tell the browser or CDN three essential pieces of information:
+- **Cacheability**: Is this content cacheable?
+- **Freshness**: If it is cacheable, how long can it be cached for?
+- **Validation**: If it is cacheable, how do I ensure that my cached version is still fresh?
+
+Headers are meant to be used either alone or together. To determine if the content is **cacheable** and **fresh**, we have:
+- `Expires` specifies an explicit expiration date and time (i.e., when precisely the content expires).
+- `Cache-Control` specifies a cache duration (i.e., how long the content can be cached in the browser relative to when it was requested).
+
+When both are specified, `Cache-Control` takes precedence.
+
+Usage of the `Cache-Control` header has increased steadily since 2019. 74% of responses on Mobile pages included the `Cache-Control` header, while 75% of responses on Desktop pages utilized the header.
+
+{{ figure_markup(
+  image="cache-control-expires.png",
+  caption="Percent of responses that set `Cache-Control` and `Expires` headers.",
+  description="TODO",
```

Comment:
description="A bar chart showing usage of `Cache-Control` and `Expires` headers. For desktop and mobile, 74.2% use `Cache-Control`, 55.6% use `Expires`, 54.9% use both, and 25.1% use neither.",